### PR TITLE
Optimised show_topics

### PR DESCRIPTION
**Pull request by bhargavvader**
Possible solution to #1006 .
@tmylk could you review?

### CODE REVIEW

File: gensim/models/ldamodel.py [hunk 1]
```diff
@@ -800,15 +800,24 @@ def show_topics(self, num_topics=10, num_words=10, log=False, formatted=True):
             chosen_topics = sorted_topics[:num_topics // 2] + sorted_topics[-num_topics // 2:]
 
         shown = []
+
+        topic = self.state.get_lambda()
         for i in chosen_topics:
             if formatted:
```
Comment (tmylk) [inline gensim/models/ldamodel.py]: Too much code duplication
Just add an extra `if formatted` block in the end that changes formatting

Comment (bhargavvader) [inline gensim/models/ldamodel.py]: Oh yes, that was quite stupid of me :shit:
Will fix!

**Comment by bhargavvader** [comment]
@tmylk , removed duplicate code.

**Comment by tmylk** [comment]
Is there a performance improvement with `%%timeit`?

**Comment by bhargavvader** [comment]
Yeah, with the old method and 5 topics for the `news_corpus` dataset it was `100 loops, best of 3: 6.89 ms per loop`, with the optimised one it was `100 loops, best of 3: 4.28 ms per loop`

**Comment by tmylk** [comment]
hmm, that doesn't agree with the original report of 

> On my setup (500 topics, 100k words, learned with 500k documents), calling show_topics takes 30sec, with 28sec inside the 500 get_lambda() calls

**Comment by bhargavvader** [comment]
Could it maybe be because of the number of topics? Will increase number of topics and see what the difference is like. I only have 5 topics.

**Comment by tmylk** [comment]
Yes, let's test the reported case

**Comment by bhargavvader** [comment]
I got a significant improvement (20 times faster) with 500 topics (vocab is smaller though, len is 44143).

For the old method:

`1 loop, best of 3: 2.05 s per loop`

For the optimised method:

`10 loops, best of 3: 107 ms per loop`

**Comment by tmylk** [comment]
Thanks a lot for the improvement!

**Comment by bhargavvader** [comment]
😄

**Pull request merged**